pagopa/io-dev-api-server
# [IOPID-1139] Refactored home dashboard & added profile custom configuration for email validation

## Short description
This pr:
- adds new management of the dev server home to be able to insert configurations in a modular way.
- adds the configuration for email validation.
- refactor the way the profile is managed.
- updates the README accordingly.

<details><summary>Details of the new dashboard section</summary>
<p>

![image](https://github.com/pagopa/io-dev-api-server/assets/16268789/837c6499-854f-4a04-88d1-814cc1a50689)

</p>
</details> 

## List of changes proposed in this pull request
- assets/html/homeDashboard.html: new home.
- assets/html/mail.html: new mail configuration option.
- src/persistence/profile.ts: logic to manage the profile data.
- src/routers/configHomeDashboard/configHomeDashboard.ts: logic to add configuration component to the dev server home.
- src/routers/profile.ts: added the logic to enable the profile custom config.
- src/routers/profileCustomConfig/mailDashboardConfig.ts: added logic to handle the custom email configuration.
- src/routers/walletsV2/configDashboard.ts: removed the handler that shows the wallet configuration in home.
- src/server.ts: added new routers.
- README.md: updated accordingly.

## How to test
- Try creating new "html components" and adding them to the homeDashboard file with the <include> tag.
- Try modifying the email validation configuration and verify that the payload of the get and post profile calls responds as you expect.
